bayesian/utils.py

```diff
@@ -329,14 +329,23 @@ def search_packages_from_graph(tokens):
     return {'result': pkg_list}
 
 
-def get_analyses_from_graph(ecosystem, package, version):
+def get_analyses_from_graph(ecosystem, package, version, vendor=None):
     """Read analysis for given package+version from the graph database."""
     script1 = """\
-g.V().has('pecosystem', ecosystem).has('pname', name).has('version', version).as('version')\
-.in('has_version').dedup().as('package').select('version').coalesce(out('has_cve')\
-.as('cve').select('package','version','cve').by(valueMap()),select('package','version')\
-.by(valueMap()));\
-"""
+        g.V().has('pecosystem', ecosystem).has('pname', name).has('version', version).as('version')\
+        .in('has_version').dedup().as('package').select('version').coalesce(out('has_cve')\
+        .as('cve').select('package','version','cve').by(valueMap()),select('package','version')\
+        .by(valueMap()));\
+        """
+
+    script2 = \
+        "g.V().has('pecosystem', ecosystem).has('pname', name)" \
+        ".has('version', version).in('has_version')" \
+        ".out('has_version').not(out('has_cve')).values('version').dedup();"
+
+    if vendor:
```

Comment:
what happens when no vendor is provided? for the current ext users? This logic needs to be changed

Reply:
if vendor is None.. Flow Continues..skipping Condition. Please relook

Reply:
i got it.  what im trying to say is, you need a separate function call where it does all the fetching etc for snyk cves.

Reply:
Added Separate Query Builder Function